webUI/react/src/components/reader/pager/VerticalPager.tsx

```diff
@@ -29,36 +29,87 @@ export default function VerticalReader(props: IReaderProps) {
     const classes = useStyles();
     const history = useHistory();
 
-    const [initialScroll, setInitialScroll] = useState(-1);
-    const initialPageRef = useRef<HTMLDivElement>(null);
+    const pageRef = useRef<HTMLDivElement>(null);
+    const pagesRef = useRef<HTMLDivElement[]>([]);
+    let readerCurPage = curPage;
+
+    function nextPage() {
+        if (curPage < pages.length - 1) {
+            setCurPage((x) => {
+                readerCurPage = x + 1;
+                return readerCurPage;
+            });
+            pagesRef.current[readerCurPage].scrollIntoView();
+        } else if (settings.loadNextonEnding) {
+            nextChapter();
+        }
+    }
+
+    function prevPage() {
+        if (readerCurPage > 0) {
+            setCurPage((page) => {
+                const rect = pagesRef.current[page].getBoundingClientRect();
+                readerCurPage = (rect.y < 0 && rect.y + rect.height > 0) ? page : page - 1;
+                return readerCurPage;
+            });
+        }
+        pagesRef.current[readerCurPage].scrollIntoView();
```

Comment:
same as the other comment.

Reply:
The reason I added an additional check in there is because if an image is being cut of by the top of the screen, it should reset the view back to top of the image.
Like this:
![prev](https://user-images.githubusercontent.com/35658388/119360871-ec75f300-bc5f-11eb-93a3-3e150ef9ca89.png)

Reply:
the comment is about readerCurPage